service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/EncounterMapper.java

```diff
@@ -53,10 +57,34 @@ public String mapEncounterToEhrComposition(Encounter encounter) {
             .displayName(buildDisplayName(encounter))
             .originalText(buildOriginalText(encounter));
 
+        final String recReference = findParticipantWithCoding(encounter, ParticipantCoding.RECORDER)
+            .map(idMapper::get)
+            .orElseThrow(() -> new EhrMapperException("Encounter.participant recorder is required"));
+
+        final Optional<String> pprfReference = findParticipantWithCoding(encounter, ParticipantCoding.PERFORMER)
+            .map(idMapper::get);
```

Comment:
#156 adds a test for the missing recorder reference; I'd like to wait until that is merged to main (and then this) before updating the tests for the missing performer. Will just make the work simpler.

Reply:
Test now added